src/azure-cli/azure/cli/command_modules/resource/_bicep.py

```diff
@@ -0,0 +1,148 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import re
+import stat
+import platform
+import subprocess
+
+from pathlib import Path
+from contextlib import suppress
+
+import requests
+import semver
+
+from six.moves.urllib.request import urlopen
+from knack.log import get_logger
+from azure.cli.core.azclierror import FileOperationError, ValidationError, UnclassifiedUserFault, ClientRequestError
+
+# See: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+_semver_pattern = r"(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?"  # pylint: disable=line-too-long
+_logger = get_logger(__name__)
+
+
+def run_bicep_command(args, auto_install=True, check_upgrade=True):
+    installation_path = _get_bicep_installation_path(platform.system())
+    installed = os.path.isfile(installation_path)
+
+    if not installed:
+        if auto_install:
+            ensure_bicep_installation()
+        else:
+            raise FileOperationError('Bicep CLI not found. Install it now by running "az bicep install".')
+    elif check_upgrade:
+        with suppress(ClientRequestError):
+            # Checking upgrade should ignore connection issues.
+            # Users may continue using the current installed version.
+            installed_version = _get_bicep_installed_version(installation_path)
+            latest_release_tag = get_bicep_latest_release_tag()
+            latest_version = _extract_semver(latest_release_tag)
+            if installed_version and latest_version and semver.compare(installed_version, latest_version) < 0:
+                _logger.warning(
+                    'A new Bicep release is available: %s. Upgrade now by running "az bicep upgrade".',
+                    latest_release_tag,
+                )
+
+    return _run_command(installation_path, args)
+
+
+def ensure_bicep_installation(release_tag=None):
+    system = platform.system()
+    installation_path = _get_bicep_installation_path(system)
+
+    if os.path.isfile(installation_path):
+        if not release_tag:
+            return
+
+        installed_version = _get_bicep_installed_version(installation_path)
+        target_version = _extract_semver(release_tag)
+        if installed_version and target_version and semver.compare(installed_version, target_version) == 0:
```

Comment:
@shenglol, if `bicep` has no fancy versioning schema, can `semver` be replaced by `packaging.version`? `packaging.version` is used by `az extension` (#17667):

https://github.com/Azure/azure-cli/blob/22725b36de51951d41be2e64e02731ff45379606/src/azure-cli-core/azure/cli/core/extension/__init__.py#L287-L298